### ISSUE TITLE
Invalid tree when the routes are added on non-deterministic way

### ISSUE THREAD

**Issue by sergiotudela** [lantunes/routd#8]
Hi,

I found a wrong behavior in a project, just wrote this following test to reproduce:

``` java
@It("produces the correct trees when the routes are loaded on non-deterministic way")
    public void treeTest21() {
        final Route r1  = new Route("/abc/:param1/:param2/:param4/def/:param5");
        final Route r2  = new Route("/abc/:param1/:param2/:param3");

        router.add(r1);
        router.add(r2);

        assertEquals(r2, router.route("/abc/dummy1/dummy2/dummy3"));
        assertEquals(r1, router.route("/abc/dummy1/dummy2/dummy3/def/dummy4"));

        assertEquals(null, router.route("/abc"));
        assertEquals(null, router.route("/abc/dummy1"));
        assertEquals(null, router.route("/abc/dummy1/dummy2"));
        assertEquals(null, router.route("/abc/dummy1/dummy2/dummy3/def"));
    }
```

The output:

```
Results :

Failed tests: 
  TestTreeRouter.treeTest21:600 expected:</abc/:param1/:param2/:param3> but was:<null>
```

**Comment by sergiotudela** [comment]
I just applied some changes to fix the issue. Take a look: https://github.com/sergiotudela/routd/commit/06452c4bd093fcd9e35cd306514d1f564645fc5e

The output,

```
-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running org.bigtesting.routd.tests.TestRouteHelper
Tests run: 8, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.044 sec - in org.bigtesting.routd.tests.TestRouteHelper
Running org.bigtesting.routd.tests.TestRegexRoute
Tests run: 16, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.017 sec - in org.bigtesting.routd.tests.TestRegexRoute
Running org.bigtesting.routd.tests.TestRegexRouter
Tests run: 34, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.034 sec - in org.bigtesting.routd.tests.TestRegexRouter
Running org.bigtesting.routd.tests.TestRoute
Tests run: 60, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.029 sec - in org.bigtesting.routd.tests.TestRoute
Running org.bigtesting.routd.tests.TestRegexRouteComparator
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 sec - in org.bigtesting.routd.tests.TestRegexRouteComparator
Running org.bigtesting.routd.tests.TestTreeRouter
Tests run: 55, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.033 sec - in org.bigtesting.routd.tests.TestTreeRouter

Results :

Tests run: 174, Failures: 0, Errors: 0, Skipped: 0
```

**Comment by lantunes** [comment]
Hi, thanks for this. Development has been slow, so I apologize for the delay. I will merge this shortly.

**Comment by sergiotudela** [comment]
No worries! Thanks!

**Comment by lantunes** [comment]
This has been fixed in: 374ebe154b0426b7495fe3427194290a67a29703
